packages/router/src/grpc/offscreen-client.ts

```diff
@@ -61,29 +62,35 @@ const buildActions = (
   fullViewingKey: FullViewingKey,
   cancel: PromiseLike<never>,
 ): Promise<Action>[] => {
-  const active = activateOffscreen();
-  // eslint-disable-next-line @typescript-eslint/no-floating-promises
+  const activation = activateOffscreen();
+
+  // this is a lot of binary -> base64 serialization, so just do it once and reuse
```

Comment:
```suggestion
  // this is a lot of JSON serialization, so just do it once and reuse
```

(Right? We're not doing binary -> base64 here, right?)

Reply:
these pb types include many binary fields, and pb json involves binary -> base64. binary -> base64 is notably slow

Reply:
Gotcha — but we're still encoding to JSON, not base64, right?

Reply:
the mentioned base64 is inside json yes. i'll update the comment